### PR TITLE
Fixed the test-path and yaml files

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/docindex.yml [hunk 1]
```diff
@@ -34,16 +35,39 @@ jobs:
...
-      - task: CopyFiles@2
-        displayName: Copy HTML to Artifacts Directory
-        inputs:
-          sourceFolder: $(Build.SourcesDirectory)/docfx_project/
-          content: '**\*'
-          targetFolder: $(Build.ArtifactStagingDirectory)/docfx_project
-          overWrite: true
+          SourceDirectory: $(Build.ArtifactStagingDirectory)
+      - pwsh: |
+          Copy-Item -Path $(Build.SourcesDirectory)/eng/common/scripts/git-branch-push.ps1 -Destination ./ -Recurse -Force
```
Comment (weshaggard) [inline eng/pipelines/docindex.yml:40]: We should consider copying all of eng directory incase we get extra dependencies or other tools we use from our normal repos.

File: eng/pipelines/docindex.yml [hunk 2]
```diff
@@ -34,16 +35,39 @@ jobs:
...
+          targetPath: $(Build.ArtifactStagingDirectory)/docfx_project/_site
+          
+      - pwsh: |
+          git checkout -b gh-pages-local --track origin/gh-pages-root
+        workingDirectory: $(Build.SourcesDirectory)
+        displayName: Git pull GH pages branch
+
+      - pwsh: |
+          Copy-Item -Path $(Build.ArtifactStagingDirectory)/docfx_project/_site/* -Destination ./ -Recurse -Force
+          git add -A
```
Comment (weshaggard) [inline eng/pipelines/docindex.yml:58]: We shouldn't need this explicit add.

File: eng/pipelines/docindex.yml (hunk 1 above)
Comment (sima-zhu) [reply eng/pipelines/docindex.yml:40]: Do you mean we add a eng/common folder to gh-pages branch?

File: eng/pipelines/docindex.yml (hunk 2 above)
Comment (sima-zhu) [reply eng/pipelines/docindex.yml:58]: All files copied to the gh-pages-local branch are untracked from git. git-branch-push will create a new branch which inherit gh-pages-local which did not include the new changes.

File: eng/pipelines/docindex.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/docindex.yml:40]: I originally thought eng/common but things like the Language-Settings.ps1 script lives under eng/scripts and so I want to be sure if we start including that in our ps scripts we don't break this. That is why I suggest copying all of the eng directory.

File: eng/pipelines/docindex.yml (hunk 2 above)
Comment (weshaggard) [reply eng/pipelines/docindex.yml:58]: Yes but when we do the commit we pass in the "-am" option with the "-a" part means include all changes.  Is that not picking up new files? If it is not the doing the "git add -A"

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/docindex.yml (hunk 2 above)
Comment (sima-zhu) [reply eng/pipelines/docindex.yml:58]: They are not the same branch. git-branch-push is on gh-pages which inherited from gh-page-local I am creating. However, the changes are untracked in gh-pages-local if no `git add -A`

**Pull request merged**